app/models/miq_server/worker_management/monitor.rb

```diff
@@ -72,42 +67,33 @@ def sync_workers
   end
 
   def clean_worker_records
-    processed_workers = []
     miq_workers.each do |w|
       next unless w.is_stopped?
       _log.info("SQL Record for #{w.format_full_log_msg}, Status: [#{w.status}] is being deleted")
-      processed_workers << w
       worker_delete(w.pid)
       w.destroy
+      miq_workers.delete(w)
```

Comment:
Modifying a collection while iterating over it feels wrong...If you're trying to return the array, it might be better to use `.reject` instead of `.each`, and then just have this be `true` (or you might not need this last line since `.destroy` will return itself, which is truthy.

Reply:
Yeah, there's a lot wrong here. Especially because we're deleting from the cached association.

@jrafanie I think just found another issue with this. What are you going to be changing in this area @jrafanie? We should probably work out a common solution.